s2/loop.go

```diff
@@ -1612,12 +1612,12 @@ func (l *loopCrosser) hasCrossing(ai, bi *rangeIterator) bool {
 	return false
 }
 
-// containsCenterMatches reports if the clippedShapes containsCenter boolean corresponds
-// to the crossing target type given. (This is to work around C++ allowing false == 0,
-// true == 1 type implicit conversions and comparisons)
-func containsCenterMatches(a *clippedShape, target crossingTarget) bool {
-	return (!a.containsCenter && target == crossingTargetDontCross) ||
-		(a.containsCenter && target == crossingTargetCross)
+// containsCenterMatches reports if the clippedShapes containsCenter boolean
+// corresponds to the crossing target type given. (This is to work around C++
+// allowing false == 0, true == 1 type implicit conversions and comparisons)
+func containsCenterMatches(a bool, target crossingTarget) bool {
```

Comment:
Why not make a  `*rangeIterator`?  That way you can ensure that the boolean means what you intend it to mean.  Alternatively, consider renaming `a` to `containsCenter`.

Reply:
renamed to containsCenter